api/src/flows.ts

```diff
@@ -413,7 +416,21 @@ class FlowManager {
 
 			return { successor: operation.resolve, status: 'resolve', data: result ?? null, options };
 		} catch (error: unknown) {
-			return { successor: operation.reject, status: 'reject', data: error ?? null, options };
+			// Is the error a JSON string? If so, parse it and use that as the error
+			const jsonError = isJSON(error)
+				? parseJSON(String(error))
+				: JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
+
+			return { successor: operation.reject, status: 'reject', data: jsonError ?? null, options };
 		}
 	}
 }
+
+function isJSON(value: any): boolean {
+	try {
+		parseJSON(value);
+		return true;
+	} catch {
+		return false;
+	}
```

Comment:
Can we extract this to a utility function so it can be re-used? We're using at least one exact copy of this function elsewhere https://github.com/directus/directus/blob/main/api/src/operations/request/index.ts#L36

Reply:
I was wanting to, but this function was inconsistently used in the code base like the following as an almost similar name but slightly different function. One returns the JSON one and one returns true.
https://github.com/directus/directus/blob/e3c755dcf000aafc6535cc271093242da5e5dbc9/api/src/env.ts#L508

What do you think the best way to name/differentiate the two is?

Reply:
As for your and the linked function i'd personally think `isValidJSON` is a proper name implying that it's a boolean value by with the word `Valid` while `tryJSON` seems to be a very basic wrapper for `parseJSON` sepcifically made for `env.ts` because it is used a couple times. No need to abstract the latter.

Reply:
The test looks good to me 😄 not sure what else you could test on this util